connection.go

```diff
@@ -878,6 +885,45 @@ func (c *Connection) openComplete() error {
 	return nil
 }
 
+// tlsConfigFromURI tries to create TLS configuration based on query parameters.
+// Returns default (empty) config in case no suitable client cert and/or client key not provided.
+// Returns error in case certificates can not be parsed.
+func tlsConfigFromURI(uri URI) (*tls.Config, error) {
+	var certPool *x509.CertPool
+	if uri.CACertFile != "" {
+		data, err := ioutil.ReadFile(uri.CACertFile)
+		if err != nil {
+			return nil, fmt.Errorf("read CA certificate: %w", err)
+		}
+
+		certPool = x509.NewCertPool()
+		certPool.AppendCertsFromPEM(data)
+	} else if sysPool, err := x509.SystemCertPool(); err != nil {
+		return nil, fmt.Errorf("load system certificates: %w", err)
+	} else {
+		certPool = sysPool
+	}
+
+	if uri.CertFile == "" || uri.KeyFile == "" {
+		// no client auth (mTLS), just server auth
+		return &tls.Config{
+			RootCAs:    certPool,
+			ServerName: uri.ServerName,
+		}, nil
+	}
+
+	certificate, err := tls.LoadX509KeyPair(uri.CertFile, uri.KeyFile)
+	if err != nil {
+		return nil, fmt.Errorf("load client certificate: %w", err)
```

Comment:
`could not load client certificate: %w`

Reply:
It's pretty common to avoid negative verb in wrapped errors to prevent duplicated messages (`load client certificate: tls: read file: file not found` instead of `could not load client certificate: tls: read file: file not found`). For example: https://github.com/uber-go/guide/blob/master/style.md#error-wrapping or https://go.dev/doc/effective_go#errors .
However, if you insist on this code style for the repo I will no argue though.